fenix/app/src/androidTest/java/org/mozilla/fenix/helpers/MatcherHelper.kt

```diff
@@ -96,15 +102,51 @@ object MatcherHelper {
     ) {
         for (appItem in appItems) {
             if (exists) {
+                Log.i(TAG, "assertUIObjectExists: Trying to verify ${appItem.selector} exists")
                 assertTrue("${appItem.selector} does not exist", appItem.waitForExists(waitingTime))
                 Log.i(TAG, "assertUIObjectExists: Verified ${appItem.selector} exists")
             } else {
+                Log.i(TAG, "assertUIObjectExists: Trying to verify ${appItem.selector} does not exist")
                 assertFalse("${appItem.selector} exists", appItem.waitForExists(waitingTimeShort))
                 Log.i(TAG, "assertUIObjectExists: Verified ${appItem.selector} does not exist")
             }
         }
     }
 
+    fun assertComposeItemExists(
+        vararg composeItems: SemanticsNodeInteraction,
+        exists: Boolean = true,
+    ) {
+        for (composeItem in composeItems) {
+            if (exists) {
+                Log.i(TAG, "assertComposeItemExists: Trying to verify compose item ${composeItem.printToString()} exists")
+                composeItem.assertExists()
```

Comment:
I think this only verifies and syncs to make sure the elements exists in the tree, not if it's actually displayed on the screen. assertIsDisplayed() should do that. Maybe we should have both.